frontend/lib/src/components/widgets/Slider/Slider.tsx

```diff
@@ -282,159 +149,294 @@ class Slider extends React.PureComponent<Props, State> {
-        "onKeyUp",
-        "onKeyDown",
-        "onMouseEnter",
-        "onMouseLeave",
-        "draggable",
-      ])
-      const ariaValueText: Record<string, string> = {}
-
-      if (this.props.element.options.length > 0 || this.isDateTimeType()) {
-        ariaValueText["aria-valuetext"] = formattedValue
+    thumbRefs.map((ref, i) => {
+      if (!ref.current) {
+        return
       }
+      ref.current.setAttribute("aria-valuetext", formattedValueArr[i])
+    })
+  })
 
-      return (
-        <StyledThumb
-          {...passThrough}
-          disabled={props.$disabled === true}
-          ref={this.thumbRef[thumbIndex]}
-          aria-valuetext={formattedValue}
-          aria-label={this.props.element.label}
-        >
-          <StyledThumbValue
-            data-testid="stSliderThumbValue"
-            disabled={props.$disabled === true}
-            ref={this.thumbValueRef[thumbIndex]}
-          >
-            {formattedValue}
-          </StyledThumbValue>
-        </StyledThumb>
-      )
+  // When resetting a form, `value` will change so we need to change `uiValue`
+  // to match.
+  useEffect(() => {
+    if (value !== uiValue) {
+      setUiValue(value)
     }
+    // Don't include `uiValue` in the deps below or the slider will become
+    // jittery.
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [value])
+
+  const debouncedSetValueWithSource = useCallback(
+    debounce(DEBOUNCE_TIME_MS, (value: number[]): void => {
+      setValueWithSource({ value, fromUi: true })
+    }),
+    [setValueWithSource]
   )
 
-  private renderTickBar = (): JSX.Element => {
-    const { disabled, element } = this.props
-    const { max, min } = element
+  const handleChange = useCallback(
+    ({ value }: { value: number[] }): void => {
+      setUiValue(value)
+      debouncedSetValueWithSource(value)
+    },
+    [setValueWithSource]
+  )
 
+  const renderTickBar = useCallback((): ReactElement => {
     return (
       <StyledTickBar data-testid="stSliderTickBar">
         <StyledTickBarItem
           disabled={disabled}
           data-testid="stSliderTickBarMin"
         >
-          {this.formatValue(min)}
+          {formattedMinValue}
         </StyledTickBarItem>
         <StyledTickBarItem
           disabled={disabled}
           data-testid="stSliderTickBarMax"
         >
-          {this.formatValue(max)}
+          {formattedMaxValue}
         </StyledTickBarItem>
       </StyledTickBar>
     )
-  }
-
-  public render(): React.ReactNode {
-    const { disabled, element, theme, width, widgetMgr } = this.props
-    const { colors, fonts, fontSizes, spacing } = theme
-    const style = { width }
-
-    // Manage our form-clear event handler.
-    this.formClearHelper.manageFormClearListener(
-      widgetMgr,
-      element.formId,
-      this.onFormCleared
-    )
+    // Only run this on first render. No real reason, just nicer.
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [])
```

Comment:
If `disabled` for example changes dynamically in the code, could this lead to an error here because the component is not updated? Do you see an issue with passing `[formattedMinValue, formattedMaxValue, disabled]`?

Reply:
Just tested it. If you have code like this:

```python
st.slider("Label 4", 10000, 25000, 10000, disabled=st.checkbox("Disable"))
```

and you check the disable box, the tickBars are not affected live and stay enabled / disabled. So we should pass `[formattedMinValue, formattedMaxValue, disabled]` here to make it work. 
Perhaps we should update our `e2e_playwright/st_slider.py` test with this^ interactivity to catch this automatically (or at least we could think about filing a tech debt item; I have the feeling that our tests do not go into much activity details like this - and maybe it would be too much overhead to test any single thing in the tests)